integration/upload_http_util.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2022 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+package integration
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+)
+
+const (
+	port        = 9001
+	httpsMethod = "https.method"
+	httpsURL    = "https.url"
+)
+
+type serveHandler struct {
+	uploadedData map[string]interface{}
+}
+
+type httpUploadHandler struct {
```

Comment:
The integration TC should not rely on such test HTTP server. For example, how-to guide is using servefile. The download URL can be provided in the related event info and can be further used to compare file content.